Framework/Nexus/src/NexusClasses.cpp

```diff
@@ -466,10 +442,17 @@ void NXDataSet::getData(void *data) {
  *   @throw runtime_error if the operation fails.
  */
 void NXDataSet::getSlab(void *data, NXDimArray const &start, NXDimArray const &size) {
-  NXopendata(m_fileID, name().c_str());
-  if (NXgetslab(m_fileID, data, &(start[0]), &(size[0])) != NXstatus::NX_OK)
-    throw std::runtime_error("Cannot read data slab from NeXus file");
-  NXclosedata(m_fileID);
+  ::NeXus::DimSizeVector vstart(start.size()), vsize(start.size());
```

Comment:
Isn't `start.size()` always 4? I think this should be `vstart(this->rank()`

Reply:
... and the for loop could probably be converted to [`std::copy_if`](https://en.cppreference.com/w/cpp/algorithm/copy) with a lambda for the condition.